gitlabform/processors/project/project_job_token_scope_processor.py

```diff
@@ -0,0 +1,186 @@
+        job_token_scope.enabled = limit_access_to_this_project
+        job_token_scope.save()
+
+    def _process_projects(
+        self,
+        project: Project,
+        job_token_scope: ProjectJobTokenScope,
+        projects_allowlist: List,
+    ):
+        if not projects_allowlist:
+            warning(
+                "Process will remove existing projects from allowlist, as none set in configuration"
+            )
+
+        existing_allowlist = job_token_scope.allowlist.list()
+
+        target_project_ids = self._get_target_project_ids_from_config(
+            projects_allowlist
+        )
+
+        project_ids_to_add = self._get_ids_to_add_to_allowlist(
+            existing_allowlist, target_project_ids
+        )
+        for project_id in project_ids_to_add:
+            if project_id != project.id:
+                job_token_scope.allowlist.create({"target_project_id": project_id})
+                job_token_scope.save()
+
+        self._remove_projects_from_allowlist(
+            project, job_token_scope, existing_allowlist, target_project_ids
+        )
+
+    def _process_groups(
+        self,
+        job_token_scope: ProjectJobTokenScope,
+        groups_allowlist: List,
+    ):
+        if not groups_allowlist:
+            warning(
+                "Process will remove existing groups from allowlist, as none set in configuration"
+            )
+
+        existing_allowlist = job_token_scope.groups_allowlist.list()
+
+        target_group_ids = self._get_target_group_ids_from_config(groups_allowlist)
+
+        group_ids_to_add = self._get_ids_to_add_to_allowlist(
+            existing_allowlist, target_group_ids
+        )
+
+        for group_id in group_ids_to_add:
+            job_token_scope.groups_allowlist.create({"target_group_id": group_id})
+            job_token_scope.save()
+
+        self._remove_groups_from_allowlist(
+            job_token_scope, existing_allowlist, target_group_ids
+        )
+
+    def _remove_groups_from_allowlist(
+        self,
+        job_token_scope: ProjectJobTokenScope,
+        existing_allowlist,
+        target_group_ids: List,
+    ):
+        group_ids_to_remove = self._get_ids_to_remove_from_allowlist(
+            existing_allowlist, target_group_ids
+        )
+        for group_id in group_ids_to_remove:
+            job_token_scope.groups_allowlist.delete(group_id)
+            job_token_scope.save()
+
+    def _remove_projects_from_allowlist(
+        self,
+        project: Project,
+        job_token_scope: ProjectJobTokenScope,
+        existing_allowlist,
+        target_project_ids: List,
+    ):
+        project_ids_to_remove = self._get_ids_to_remove_from_allowlist(
+            existing_allowlist, target_project_ids
+        )
+        for project_id in project_ids_to_remove:
+            if project_id != project.id:
+                job_token_scope.allowlist.delete(project_id)
```

Comment:
Similar comment as above regarding method name. I feel `remove` is probably more appropriate than `delete`.